tests/test_client.py

```diff
@@ -395,6 +397,28 @@ def test_post_records(self):
         self.client.post_records("myCollection", records)
 
 
+class HandleSyncRequestsResponseTest(unittest.TestCase):
+    def setUp(self):
+        super(HandleSyncRequestsResponseTest, self).setUp()
+        self.client = SyncClient(
+            hashalg=mock.sentinel.hashalg,
+            id=mock.sentinel.id,
+            key=mock.sentinel.key,
+            uid=mock.sentinel.uid,
+            api_endpoint="http://sync.services.mozilla.com/"
+        )
+
+    def test_get_record_can_handle_empty_response(self):
+        with mock.patch("syncclient.client.requests.request") as request:
+            response = mock.MagicMock()
+            response.status_code = 304
+            response.response = "Not Modified"
+            response.url = "http://www.example.com/"
+            request.return_value = response
+            self.assertRaises(HTTPError,
```

Comment:
> as we do it in other parts of the tests

There are many places where we are using assertRaises in this specific project. I am opening #15 to convert tests assertion to py.test based ones.